sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousRateStreamSource.scala

```diff
@@ -36,6 +38,9 @@ class RateStreamContinuousStream(rowsPerSecond: Long, numPartitions: Int) extend
 
   val perPartitionRate = rowsPerSecond.toDouble / numPartitions.toDouble
 
+  val highestCommittedValue = new AtomicLong(Long.MinValue)
+  val firstCommittedTime = new AtomicLong(Long.MinValue)
```

Comment:
So, do we need to add  these variable only for testing? Is there any other valuable information about these?

Reply:
Actually yes, and `creationTime` is also only used from tests. Rate stream is designed for test and evaluation purpose, so it seems OK to modify rate stream to support tests.

Btw, I'll adjust the scope to `private[sql]`, as it doesn't need to be exposed outside of Spark.